client/lang/py/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+use image;
+use numpy::PyReadonlyArrayDyn;
+use pyo3::prelude::*;
+use pythonize::depythonize;
+use serde_json::Value as JsonValue;
+use sha2::Digest;
+use std::collections::HashMap;
+
+use ::wandbinder::{run, session, settings, wandb};
+
+fn normalize(data: &Vec<f64>) -> Vec<f64> {
+    let min = data
+        .iter()
+        .cloned()
+        .min_by(|a, b| a.partial_cmp(b).unwrap())
+        .unwrap();
+    let max = data
+        .iter()
+        .cloned()
+        .max_by(|a, b| a.partial_cmp(b).unwrap())
+        .unwrap();
+
+    data.iter()
+        .map(|&value| (value - min) / (max - min))
+        .collect()
+}
+
+fn ndarray_to_image(arr: PyReadonlyArrayDyn<'_, f64>, path: &String) -> run::Media {
+    let shape = arr.shape();
+    // Convert the ndarray to a Vec<f64> for serialization
+    let vec_data: Vec<f64> = arr.as_slice().unwrap().to_vec();
+    // convert to Vec<u8> for image serialization
+    let normalized = normalize(&vec_data);
+    let byte_values: Vec<u8> = normalized.iter().map(|&v| (v * 255.0) as u8).collect();
+
+    // compute sha256 of the image
+    let mut hasher = sha2::Sha256::new();
+    hasher.update(&byte_values);
+    let image_sha256 = hasher.finalize();
+    let image_sha256_str = format!("{:x}", image_sha256);
+
+    let img: image::ImageBuffer<image::Rgb<u8>, Vec<u8>> =
+        image::ImageBuffer::from_vec(shape[0] as u32, shape[1] as u32, byte_values).unwrap();
+
+    std::fs::create_dir_all(format!("{}/media/images", path)).unwrap();
+    // You can now save or manipulate the ImageBuffer
+    // use sha256 as the filename.png
+    let image_path = format!("media/images/{}.png", &image_sha256_str[..20]);
+    let full_path = format!("{}/{}", path, image_path);
+    img.save(&full_path).unwrap();
+
+    run::Media {
+        _type: "image-file".to_string(),
+        path: image_path.to_string(),
+        sha256: image_sha256_str.to_string(),
+    }
```

Comment:
The pattern I'm imagining is defining serde_json::Object types with a "_type" just looking for them in the generic rust library...